luigi/contrib/ftp.py

```diff
@@ -131,14 +131,12 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
-        path_parts = path.split('/')
-        path = '/'.join(path_parts[:-1])
-        fn = path_parts[-1]
+        dirname, fn = os.path.split(path)
 
-        files = self.conn.nlst(path)
+        files = self.conn.nlst(dirname)
 
         exists = False
-        if fn in files:
+        if path in files or fn in files:
```

Comment:
I prefer the idea of just checking the first index of `files`, seeing if it contains `'/'` and then checking for file existence accordingly. Thus, we'd have theta(n)+1 for this part.